scripts/tdd.js

```diff
@@ -32,8 +40,9 @@ snippetFiles
     return fileName;
   })
   .map(fileName => {
+    const activeOrArchive = snippetFilesActive.includes(fileName) ? SNIPPETS_ACTIVE : SNIPPETS_ARCHIVE;
     // Grab snippetData
-    const fileData = fs.readFileSync(`${SNIPPETS_PATH}/${fileName}.md`, 'utf8');
+    const fileData = fs.readFileSync(`${activeOrArchive}/${fileName}.md`, 'utf8');
     // Grab snippet Code blocks
     const fileCode = fileData.slice(fileData.indexOf('```js'), fileData.lastIndexOf('```') + 3);
```

Comment:
`fileData.indexOf('```js')`  will fail for ` ```   js`  case. Notice the extra space. I think you should match it with regex? 
Something like `/```\s*js/`

Reply:
@kriadmin ` ``` js ` is incorrect markdown (based on our guidelines) and all snippets should conform to the guidelines and snippet_template structure. This is why we review snippets before merging, in order to avoid such problems.

Reply:
@Chalarangelo I don't think ` ``` js` is incorrect because I have used it in many snippets and I am sure about it. Even the module-maker.js takes care of that `edge-case`

Reply:
@karamarimo Even though technically it is correct markdown, it does not conform to the guidelines of the repo and the template. The main reason we have those is for people who work on scripts to have a consistent base to build on, instead of handling all of these edge-cases.

Reply:
@Chalarangelo Ok as you wish. But whatever number of guidelines you write people will still break the rule.
_btw my name is @kriadmin not @karamarimo_

Reply:
@kriadmin Well, I haven't had enough copy yet to see that Github is autocompleting the wrong username! 🤣 

People constantly submit snippets with imperfections in regards to guidelines, that's why we review and modify them to save us the hassle of dealing with those tiny little details later down the line. Although, admittedly things fall through the cracks quite often.

Reply:
@Chalarangelo Also If you do not handle the edge-cases we will be lacking consistency in our scripts. The `npm-module-make.js` we have (I don't remember the name tbh) uses regex and handles the edge cases 😕

Reply:
@kriadmin I'm taking this approach due to the fact all snippets have to adhere to our guideline & snippet structure as stated by @Chalarangelo 

> ` ``` js` is incorrect markdown (based on our guidelines) and all snippets should conform to the > guidelines and snippet_template structure

Please keep in mind that this is the case only for <strong>*our project*</strong> & as such there are some certainties we are sure off. Like only coming accross ` ```js ` instead of ` ``` js` due to the structure & outlines brought about by our project

Although I do agree that ` ``` js ` is considered markdown. In this project it's is not acceptable. The same way `Array(n).fill(val)` is considered one way to initialize an array in JS. In our project it is not seen or accepted as such with regards to our snippets/project guidelines.

Reply:
@kingdavidmartins I believe this is _our_ code https://github.com/Chalarangelo/30-seconds-of-code/blob/master/scripts/module.js#L20 . But it is handling the edge case

Reply:
@kingdavidmartins Run a preliminary check on existing snippets, see if any code blocks are not properly fenced as stated in guidelines and fix them, if you find them. From here on out, we keep guidelines more strict on this one point in order to make sure no scripting breaks (especially not tdd). Also, run a regex like we do in the [module](https://github.com/Chalarangelo/30-seconds-of-code/blob/master/scripts/module.js#L20) script just in case we miss something and the whole thing comes crashing down.

Reply:
@Chalarangelo Wrote a script to check if any snippet has ` ```js ` and can now confirm none of our scripts have ` ```js `  & updated tdd to search code blocks with regex